pkg/kubelet/kubelet_pods.go

```diff
@@ -1165,7 +1168,7 @@ func (kl *Kubelet) HandlePodCleanups(ctx context.Context) error {
 
 	// Remove orphaned pod statuses not in the total list of known config pods
 	klog.V(3).InfoS("Clean up orphaned pod statuses")
-	kl.removeOrphanedPodStatuses(allPods, mirrorPods)
+	kl.removeOrphanedPodStatuses(allPods, mirrorPods, possiblyRunningPods)
```

Comment:
Just to understand, it sounds like before without this change we were calling `removeOrphanedPodStatuses` on pods that were actively terminating? By passing in `possiblyRunningPods` we ensure that the status of pods are terminating is not cleared? Is it correct?

Reply:
This is probably not correct.  possiblyRunningPods includes both replaced static pods AND force deleted pods that are terminating.  The former should be in this list (well, as long as the mirror pod hasn't already been updated to change UIDs), but the latter must not be.

Reply:
I'm modulating my last comment - I think we have to describe what the desired behavior is more clearly in the proposed PR change before we can take this step.